xwiki-rendering-macros/xwiki-rendering-macro-toc/src/main/java/org/xwiki/rendering/internal/macro/toc/TocMacro.java

```diff
@@ -123,7 +130,15 @@ public boolean supportsInlineMode()
 
         TreeParametersBuilder builder = new TreeParametersBuilder();
         TreeParameters treeParameters = builder.build(rootBlock, parameters, context);
-        return this.tocTreeBuilder.build(treeParameters);
+        List<Block> parametersList = this.tocTreeBuilder.build(treeParameters);
+        if (parameters.getBox()) {
+            BoxMacroParameters boxParameters = new BoxMacroParameters();
+            boxParameters.setCssClass(context.getId() + "message");
+            parametersList = boxMacro.execute(boxParameters, content, context);
```

Comment:
@tmortagne Is it possible to reuse the BoxMacro here? How would I go about setting the parameters?